content/academy/api_scraping/general_api_scraping/handling_pagination.md

```diff
@@ -0,0 +1,205 @@
+
+```JavaScript
+// continue making requests until either we've reached 100+ items
+while (items.flat().length < 100) {
+    // if the "next_href" wasn't present in the last call, there
+    // are no more pages. return what we have and stop paginating.
+    if (!nextHref) return items.flat();
+
+    // continue paginating
+}
+```
+
+All that's left to do now is flesh out this `while` loop with pagination logic and finally return the **items** array once the loop has finished.:
+
+```JavaScript
+// index.js
+const scrapeClientId = require('./scrapeClientId');
+const { gotScraping } = require('got-scraping');
+
+const scrape100Items = async () => {
+    let nextHref = 'https://api-v2.soundcloud.com/users/141707/tracks?limit=20&offset=0';
+    let items = [];
+
+    const clientId = await scrapeClientId();
+
+    while (items.flat().length < 100) {
+        if (!nextHref) return items.flat();
+
+        // set the "client_id" URL parameter of the
+        // nextHref URL
+        const nextURL = new URL(nextHref);
+        nextURL.searchParams.set('client_id', clientId);
+
+        // make the paginated request and push its results
+        // into the in-memory "items" array
+        const res = await gotScraping(nextURL.toString());
+        const json = JSON.parse(res.body);
+        items.push(json.collection);
+
+        // queue the next link for the next loop iteration
+        nextHref = json.next_href;
+    }
+
+    // return an array of all our scraped items
+    // once the loop has finished
+    return items.flat();
+};
+
+// test run
+(async () => {
+    // run the function
+    const data = await scrape100Items();
+
+    // log the length of the items array
+    console.log(data.length);
+})();
+```
+
+> We are using the [`.flat()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) method when returning the **items** array to turn our array of arrays into a single array of items.
+
+Here's what the output of this code looks like:
+
+```text
+105
+```
+
+## [](#next) Next up
+
+<!-- In this lesson, you learned about how to use API parameters and properties returned in an API response to paginate through results. [Next up](link api_scraping/general_api_scraping/using_api_filters.md), you'll gain a solid understanding of using API filtering parameters. -->
+
+This is the last lesson in the API scraping tutorial for now, but be on the lookout for more lessons soon to come! So far, you've learned how to:
+
+1. Locate API endpoints
+2. Understand located API endpoints and their parameters
+3. Parse and modify cookies
+4. Modify/set headers
+5. Farm API tokens using Puppeteer
+6. Use paginated APIs
+<!-- 7. Utilize API filters to narrow down results -->
+
+If you'd still like to read more API scraping, check out the [**GraphQL scraping**]({{@link api_scraping/graphql_scraping.md}}) course! GraphQL is the king of API scraping.
```

Comment:
I would at least add a small section about the limited pagination and add a link to https://docs.apify.com/tutorials/scrape-paginated-sites before we create any content for this